gaia-fast-list.js

```diff
@@ -947,8 +860,15 @@ Internal.prototype = {
     if (!this.getItemImageSrc) return;
     var img = el[keys.img];
     if (!img) return;
+
+    // Hide image instantly
     img.style.transition = 'none';
     img.style.opacity = 0;
+
+    // Clear any pending callbacks
+    if (img.raf) cancelAnimationFrame(img.raf);
+    img.onload = null;
+    img.raf = null;
```

Comment:
we already stub `requestAnimationFrame`, we can stub `cancelAnimationFrame` too, not yield `requestAnimationFrame`, scroll, and check that stuff gets canceled